doc/source/examples/incompressible-flow/2d-backward-facing-step/2d-backward-facing-step.rst

```diff
@@ -0,0 +1,348 @@
+
+.. image:: image/backward_facing_step_description.png
+
+The backward-facing step problem is a classical computational fluid dynamics problem. The fact that it features a non-trivial solution while maintaining simple geometry and boundary conditions makes this problem a good candidate for validation purposes as well as to test the robustness of a given CFD method. First, the basic parameters used to solve the backward-facing step problem will be exposed. A solution to several Reynolds numbers (from :math:`Re = 100` to :math:`Re =1000`) will then be presented and compared to experimental and numerical data. A mesh adaptation and numerical error analysis will be carried on.
+
+Parameter File
+--------------
+
+The following subsections show the different parameters of *Lethe* used in the simulations. While they all remain more or less the same throughout the various cases, some of them change as the Reynolds number is increased.
+
+Simulation control
+~~~~~~~~~~~~~~~~~~
+
+For :math:`Re < 700`, the solution is stable enough to be computed in steady state by setting ``method`` to ``steady`` : 
+
+.. code-block:: text
+
+	subsection simulation control
+	  set method = steady
+	  set number mesh adapt = 10
+	  set output name = backward_facing_step_output
+	  set output frequency = 1
+	  set subdivision = 1
+	  set output boundaries = false
+	end
+	
+A mesh refinement analysis can be done with ``set number mesh adapt = 10``. By starting from a very coarse mesh and by dynamically refining the mesh at least 10 times, asymptotic convergence can be clearly observed.
+
+However, for :math:`Re \geq 700`, convergence can be quite difficult to obtain while doing a steady state simulation. In fact, as the Reynolds number increases, the problem becomes progressively stiffer to a point where the ``steady`` solver ultimately fails. With that in mind, the case can be solved as a transient problem until the steady state solution is obtained. This can be achieved with the ``method = steady_bdf`` parameter.
+
+.. code-block:: text
+
+	subsection simulation control
+	  set method = steady_bdf
+	  set stop tolerance = 1e-6
+	  set time step                     = 0.005
+	  set adapt                         = true
+	  set max cfl                       = 1e6
+	  set adaptative time step scaling = 1.2
+	  set number mesh adapt = 0
+	  set output name = backward_facing_step_output
+	  set output frequency = 1
+	  set subdivision = 1
+	  set output boundaries = false
+	end
+  
+``stop tolerance``, ``time step``, ``adapt``, ``max cfl`` and ``adaptive time step scaling`` are parameters that control the pseudo-steady simulation. Since this problem can be quite demanding, a relatively high tolerance has been chosen as an input for ``stop tolerance``. This choice is however not without its consequences : the :math:`x_r` parameter might be slightly underestimated. 
```

Comment:
Have you checked if changing this parameter changed x_r? 1e-6 is actually already quite low.